src/stateless_validation.cpp

```diff
@@ -1,23 +1,57 @@
+#include <time.h>
 
-
+#include <veriblock/consts.hpp>
 #include <veriblock/entities/hashes.hpp>
 #include <veriblock/stateless_validation.hpp>
 
 namespace VeriBlock {
 
 bool checkProofOfWork(const BtcBlock& block) {
   Sha256Hash blockHash = block.getBlockHash();
-  Sha256Hash target = decodeBits(block.bits);
+  Blob<SHA256_HASH_SIZE> target = decodeBits<SHA256_HASH_SIZE>(block.bits);
+
+  return target > blockHash;
+}
+
+bool checkProofOfWork(const VbkBlock& block) {
+  VBlakeBlockHash blockHash = block.getBlockHash();
+  Blob<VBLAKE_BLOCK_SIZE> target =
+      decodeBits<VBLAKE_BLOCK_SIZE>(block.difficulty);
 
   return target > blockHash;
 }
 
+template <typename BlockType>
+bool checkMaximumDrift(const BlockType& block) {
+  uint32_t currentTime = (uint32_t)time(0);
```

Comment:
create a file `veriblock/time.hpp` and create a function which returns current timestamp.
and use that func there.

better name it something like `currentTimestamp4` (stands for 4 bytes, because we could have another func which returns 8 bytes)